modules/passkey/test/userstories/PasskeyCredentialCreation.spec.ts

```diff
@@ -0,0 +1,142 @@
+ * Step 3: Execute the transaction.
+ */
+describe('Passkey Credential Creation for Safe Ownership [@userstory]', () => {
+  const setupTests = deployments.createFixture(async ({ deployments }) => {
+    const { SafeProxyFactory, SafeL2, FCLP256Verifier, SafeWebAuthnSignerFactory, CompatibilityFallbackHandler } = await deployments.run()
+    const [user] = await ethers.getSigners()
+
+    const proxyFactory = await ethers.getContractAt(SafeProxyFactory.abi, SafeProxyFactory.address)
+    const singleton = await ethers.getContractAt(SafeL2.abi, SafeL2.address)
+    const fallbackHandler = await ethers.getContractAt(CompatibilityFallbackHandler.abi, CompatibilityFallbackHandler.address)
+    const signerFactory = await ethers.getContractAt('SafeWebAuthnSignerFactory', SafeWebAuthnSignerFactory.address)
+    const verifier = await ethers.getContractAt('IP256Verifier', FCLP256Verifier.address)
+    const verifierAddress = await verifier.getAddress()
+
+    const navigator = {
+      credentials: new WebAuthnCredentials(),
+    }
+
+    // Create the credentials for Passkey.
+    const credential = navigator.credentials.create({
+      publicKey: {
+        rp: {
+          name: 'Safe',
+          id: 'safe.global',
+        },
+        user: {
+          id: ethers.getBytes(ethers.id('chucknorris')),
+          name: 'chucknorris',
+          displayName: 'Chuck Norris',
+        },
+        challenge: ethers.toBeArray(Date.now()),
+        pubKeyCredParams: [{ type: 'public-key', alg: -7 }],
+      },
+    })
+
+    // Get the publicKey from the credential and create the signer.
+    const publicKey = decodePublicKey(credential.response)
+    await signerFactory.createSigner(publicKey.x, publicKey.y, verifierAddress)
+    const signerAddress = await signerFactory.getSigner(publicKey.x, publicKey.y, verifierAddress)
+    const signer = await ethers.getContractAt('SafeWebAuthnSigner', signerAddress)
+
+    // Deploy Safe with the WebAuthn signer as a single owner.
+    const singletonAddress = await singleton.getAddress()
+    const setupData = singleton.interface.encodeFunctionData('setup', [
+      [signerAddress],
+      1,
+      ethers.ZeroAddress,
+      '0x',
+      await fallbackHandler.getAddress(),
+      ethers.ZeroAddress,
+      0,
+      ethers.ZeroAddress,
+    ])
+    const safeAddress = await proxyFactory.createProxyWithNonce.staticCall(singletonAddress, setupData, 0)
+    await proxyFactory.createProxyWithNonce(singletonAddress, setupData, 0)
+    const safe = await ethers.getContractAt([...SafeL2.abi, ...CompatibilityFallbackHandler.abi], safeAddress)
+
+    const { chainId } = await ethers.provider.getNetwork()
+    const domainSeparator = ethers.keccak256(
+      ethers.AbiCoder.defaultAbiCoder().encode(['bytes32', 'uint256', 'address'], [DOMAIN_SEPARATOR_TYPEHASH, chainId, safeAddress]),
+    )
```

Comment:
We should use `ethers` utilities for this IMO.

Reply:
Resolved based on previous comment.